src/Symfony/Component/HttpKernel/BootingKernel.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel;
+
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\Exception\BadMethodCallException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Bundle\BootingBundle;
+use Symfony\Component\HttpKernel\Bundle\BundleInterface;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+class BootingKernel implements KernelInterface
```

Comment:
it's a kernel decorator, so should we call it `KernelDecoratorKernel`? To me looks ugly and I wouldn't name a class after a pattern. `ImmutableKernel` wouldn't make sense if other kernel were immutable as well, it's just a way to implement something it does not express what is specific about this class. `BootingKernel` makes more sense